### PR TITLE
not following symlinks for cp_r

### PR DESCRIPTION
**Pull request by rocketnik**

### CODE REVIEW

**Comment by rocketnik** [comment]
Problem is, the changes require directory-1.3.1, but the latest lts contains 1.3.0. I will try to implement these helper functions myself within Shelly.hs

File: src/Shelly.hs [hunk 1]
```diff
@@ -1415,17 +1415,25 @@ cp_r from' to' = do
 -- | Copy a file. The second path could be a directory, in which case the
 -- original file name is used, in that directory.
 cp :: FilePath -> FilePath -> Sh ()
-cp from' to' = do
+cp = cp' True
+
+cp' :: Bool -> FilePath -> FilePath -> Sh ()
```
[the remark this answers is not in the document]
Comment (rocketnik) [reply src/Shelly.hs]: will fix

**Comment by rocketnik** [comment]
Hello Greg, I added the code from directory-1.3.1 to this project, so that the project depends on an earlier version (directory-1.3.0), which is included at least in the latest lts (11.9) - the directory-1.3.1 is not part of any lts yet. This of course breaks your tests for the older lts. I am not sure what to do about it. From my naive perspective, I would say, we should embrace the future and drop support for the old lts. Somebody using an old version lts, and with such also old versions of all the other libraries, can still use an old version of Shelly. But I am very new to the stackage idea and have no clue, what I am talking about. Hoping for your wiser view.

**Comment by gregwebs** [comment]
This looks great! The only thing to make it better would be to move these directory functions into a separate module.

**Comment by rocketnik** [comment]
Will move the directory functions into a separate file. We can later remove the file, when we can upgrade to directory-1.3.1 that will probably ship in approx. 6 month with the next lts.

**Comment by rocketnik** [comment]
Done. Please check again.
I did not move `cp_should_follow_symlinks`, because I think, it is more convenient to have it next to the cp and cp_r functions, that make use of it. But if you want to, I can also move it into Shelly.Directory. Just let me know.

**Comment by rocketnik** [comment]
Hello Greg, the tests look good. If you want to, you can review my PR again now. Thank you!

**Pull request merged**